templates/terraform/post_create/sshkeyfingerprint.go.erb

```diff
@@ -0,0 +1,15 @@
+
+loginProfile, ok := res["loginProfile"]
+if !ok {
+  return fmt.Errorf("Create response didn't contain critical fields. Create may not have succeeded.")
+}
+
+// `fingerprint` is autogenerated from the api so needs to be set post-create
+sshPublicKeys := loginProfile.(map[string]interface{})["sshPublicKeys"]
+for _,sshPublicKey := range sshPublicKeys.(map[string]interface{}) {
+  if sshPublicKey.(map[string]interface{})["key"].(string)== d.Get("key") {
+    d.Set("fingerprint", sshPublicKey.(map[string]interface{})["fingerprint"].(string))
+    break
+  }
+}
+
```

Comment:
Can you add this chunk of code here? https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-3221-old..auto-pr-3221#diff-1479abaa57e56e222e114bdb5e888531R98-R103

`fingerprint` wasn't available before, but will be now. That should properly set it in the id.

Reply:
Thanks work well